packages/database/src/migrations/migrationData/20200826215113-UseIndicatorsInStriveVisualisations-modifies-data.js

```diff
@@ -0,0 +1,234 @@
+    config: {
+      formula: 'SSWT1021 + SSWT1022 + SSWT1023',
+      aggregation: {
+        SSWT1021: 'FINAL_EACH_WEEK',
+        SSWT1022: 'FINAL_EACH_WEEK',
+        SSWT1023: 'FINAL_EACH_WEEK',
+      },
+    },
+  },
+  mrdtPositiveConsultationsAgainstTests: {
+    code: 'MRDT_POS_CONSULT_AGAINST_TESTS',
+    builder: 'arithmetic',
+    config: {
+      formula: 'MRDT_POS_CONSULT_WEEKLY / SSWT1072',
+      aggregation: {
+        MRDT_POS_CONSULT_WEEKLY: 'SUM',
+        SSWT1072: ['FINAL_EACH_WEEK', 'SUM'],
+      },
+    },
+  },
+  mrdtPositiveConsultationsAgainstTestsWeekly: {
+    code: 'MRDT_POS_CONSULT_AGAINST_TESTS_WEEKLY',
+    builder: 'arithmetic',
+    config: {
+      formula: 'MRDT_POS_CONSULT_WEEKLY / SSWT1072',
+      aggregation: {
+        MRDT_POS_CONSULT_WEEKLY: 'RAW',
+        SSWT1072: 'FINAL_EACH_WEEK',
+      },
+    },
+  },
+  mrdtPositiveConsultationsRatioWeekly: {
+    code: 'MRDT_POS_CONSULT_RATIO_WEEKLY',
+    builder: 'arithmetic',
+    config: {
+      formula: 'MRDT_POS_CONSULT_WEEKLY / SSWT1001',
+      aggregation: {
+        MRDT_POS_CONSULT_WEEKLY: 'FINAL_EACH_WEEK',
+        SSWT1001: 'FINAL_EACH_WEEK',
+      },
+    },
+  },
+};
+
+export const reportChanges = [
+  {
+    id: 'PG_Strive_PNG_Weekly_Percentage_of_Positive_Consultations',
+    new: {
+      dataBuilder: 'analytics',
+      dataBuilderConfig: {
+        dataElementCodes: [indicators.mrdtPositiveConsultationsRatioWeekly.code],
+      },
+    },
+    old: {
+      dataBuilder: 'composePercentagesPerPeriod',
+      dataBuilderConfig: {
+        percentages: {
+          value: { numerator: 'mrdtPositive', denominator: 'consultations' },
+        },
+        dataBuilders: {
+          mrdtPositive: {
+            dataBuilder: 'sumPerPeriod',
+            dataBuilderConfig: {
+              dataClasses: { value: { codes: ['SSWT1021', 'SSWT1022', 'SSWT1023'] } },
+              aggregationType: 'FINAL_EACH_WEEK',
+            },
+          },
+          consultations: {
+            dataBuilder: 'sumPerPeriod',
+            dataBuilderConfig: {
+              dataClasses: { value: { codes: ['SSWT1001'] } },
+              aggregationType: 'FINAL_EACH_WEEK',
+            },
+          },
+        },
+      },
+    },
+  },
+  {
+    id: 'PG_Strive_PNG_Weekly_Percentage_of_Positive_Malaria_Against_Consultations',
+    new: {
+      dataBuilder: 'composeDataPerPeriod',
+      dataBuilderConfig: {
+        dataBuilders: {
+          positive: {
+            dataBuilder: 'analyticsPerPeriod',
+            dataBuilderConfig: {
+              dataElementCodes: [indicators.mrdtPositiveConsultationsAgainstTestsWeekly.code],
+              aggregationType: 'FINAL_EACH_WEEK',
+            },
+          },
+          consultations: {
+            dataBuilder: 'analyticsPerPeriod',
+            dataBuilderConfig: {
+              dataElementCodes: ['SSWT1001'],
+              aggregationType: 'FINAL_EACH_WEEK',
+            },
+          },
```

Comment:
You may notice that this report (and also `PG_Strive_PNG_Weekly_Percentage_of_Positive_Consultations`) used "sum" logic to calculate their denominator. I believe it's more accurate to use a "final per period" logic, since this is the one we also use in other corresponding reports.

It seems that the survey submitters generally take care to submit one survey per facility/week, so most probably we won't see any different results in the reports.